app/autosynccontroller.cpp

```diff
@@ -8,33 +8,34 @@
  ***************************************************************************/
 
 #include "autosynccontroller.h"
+#include "coreutils.h"
 
-AutosyncController::AutosyncController( QObject *parent ) : QObject( parent )
-{
-  // raise NotImplemented :)
-}
-
-AutosyncController::~AutosyncController()
-{
-  // raise NotImplemented :)
-}
-
-AutosyncController::SyncStatus AutosyncController::syncStatus()
-{
-  return mSyncStatus;
-}
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
 
-void AutosyncController::setActiveProject( Project )
+AutosyncController::AutosyncController(
+  QgsProject *openedQgsProject,
+  QObject *parent
+)
+  : QObject( parent )
+  , mQgsProject( openedQgsProject )
 {
-  // raise NotImplemented :)
-}
+  if ( !mQgsProject )
+  {
+    CoreUtils::log( QStringLiteral( "Autosync" ), QStringLiteral( "Received an invalid active project data" ) );
+    return;
+  }
 
-void AutosyncController::setActiveQgsProject( QgsProject * )
-{
-  // raise NotImplemented :)
+  // Register for data change of project's vector layers
+  const QMap<QString, QgsMapLayer *> layers = mQgsProject->mapLayers( true );
+  for ( const QgsMapLayer *layer : layers )
+  {
+    const QgsVectorLayer *vecLayer = qobject_cast<const QgsVectorLayer *>( layer );
```

Comment:
do we need all layers or only editable? wondering if you have some OSM vector layer bg map with 100x layers if we need to register all here ?

Reply:
Good idea! Like this? https://github.com/lutraconsulting/input/pull/1992/commits/e6d292b28afd67d4bdfc32fa53b1e2765051be05